src/bindgen/parser.rs

```diff
@@ -533,33 +534,23 @@ impl Parse {
                     self.load_syn_ty(crate_name, mod_cfg, item);
                 }
                 syn::Item::Impl(ref item_impl) => {
-                    let has_assoc_const = item_impl
-                        .items
-                        .iter()
-                        .any(|item| matches!(item, syn::ImplItem::Const(_)));
-                    if has_assoc_const {
-                        impls_with_assoc_consts.push(item_impl);
-                    }
-
-                    if let syn::Type::Path(ref path) = *item_impl.self_ty {
-                        if let Some(type_name) = path.path.get_ident() {
-                            for method in item_impl.items.iter().filter_map(|item| match item {
-                                syn::ImplItem::Fn(method) if !method.should_skip_parsing() => {
-                                    Some(method)
-                                }
-                                _ => None,
-                            }) {
-                                self.load_syn_method(
-                                    config,
-                                    binding_crate_name,
-                                    crate_name,
-                                    mod_cfg,
-                                    &Path::new(type_name.unraw().to_string()),
-                                    method,
-                                )
-                            }
+                    let mut assoc_const_find = false;
+                    let mut assoc_method_find = false;
+
+                    item_impl.items.iter().any(|item| {
+                        if matches!(item, syn::ImplItem::Const(_)) && !assoc_const_find {
+                            impls_with_assoc_consts.push(item_impl);
+                            assoc_const_find = true;
+                        } else if matches!(item, syn::ImplItem::Fn(_))
+                            && !assoc_method_find
+                            && matches!(*item_impl.self_ty, syn::Type::Path(_))
+                        {
+                            impls_with_assoc_fn.push(item_impl);
+                            assoc_method_find = true;
                         }
-                    }
+
+                        assoc_const_find && assoc_method_find
```

Comment:
I'm confused, doesn't this stop once we've found both an associated constant and an associated method? Why? Shouldn't we iterate over all the items? This needs comments.